tos_test.go

```diff
@@ -0,0 +1,55 @@
+package tchannel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Two return value shim
+func makeAsOne(a, b interface{}) []interface{} {
+	return []interface{}{a, b}
+}
+
+func TestAssuredFowardValues(t *testing.T) {
```

Comment:
typically tests in Go use test tables, so this would look like:
```go

tests := []struct{
  str string
  want int
}{
  {"AF11", 10},
  {"AF12", 12},
  [...]
}

for _, tt := range tests {
  got, err := GetTosField(tt.str)
  require.NoError(t, err, "Failed to get ToS field for %v", tt.str)
  assert.Equal(t, tt.want, got, "Unexpected ToS value for %v", tt.str)
}
```